presto-hive-metastore/src/test/java/com/facebook/presto/hive/metastore/thrift/MockHiveMetastoreClient.java

```diff
@@ -162,13 +168,13 @@ public Table getTable(String dbName, String tableName)
         if (throwException) {
             throw new RuntimeException();
         }
-        if (!dbName.equals(TEST_DATABASE) || (!tableName.equals(TEST_TABLE) && !tableName.equals(TEST_TABLE_WITH_CONSTRAINTS))) {
```

Comment:
We can add this check back once we make the other suggested changes as discussed.

Reply:
This cannot be added back because this will result in returning Optional.empty() from getTable() call and the testcase `testInvalidGetPartitionNames` calls getPartitionNames of CachingHiveMetastore, where HiveTableName object needs to be created as the caching key. Hence we need to get databaseName and tableName from `Optional<Table>` at this point.

Reply:
As discussed offline, we can make changes to the CachingKey for partitionNamesCache to make use of the table object rather than HiveTableNameKey since we have the table object now as part of the method arguments.